contrib/arrow-cmake/CMakeLists.txt

```diff
@@ -303,72 +339,48 @@ set(ARROW_SRCS
-
-target_link_libraries(${ARROW_LIBRARY} PRIVATE ${FLATBUFFERS_LIBRARY} boost::filesystem)
-
-if (USE_INTERNAL_PROTOBUF_LIBRARY)
-    add_dependencies(${ARROW_LIBRARY} protoc)
-endif ()
-
-target_include_directories(${ARROW_LIBRARY} SYSTEM PUBLIC ${ARROW_SRC_DIR})
-target_include_directories(${ARROW_LIBRARY} SYSTEM PUBLIC "${CMAKE_CURRENT_BINARY_DIR}/cpp/src")
-target_link_libraries(${ARROW_LIBRARY} PRIVATE ${DOUBLE_CONVERSION_LIBRARIES} ${Protobuf_LIBRARY})
-target_link_libraries(${ARROW_LIBRARY} PRIVATE lz4)
-if (ARROW_WITH_SNAPPY)
-    target_link_libraries(${ARROW_LIBRARY} PRIVATE ${SNAPPY_LIBRARY})
-endif ()
-if (ARROW_WITH_ZLIB)
-    target_link_libraries(${ARROW_LIBRARY} PRIVATE ${ZLIB_LIBRARIES})
-endif ()
-if (ARROW_WITH_ZSTD)
-    target_link_libraries(${ARROW_LIBRARY} PRIVATE ${ZSTD_LIBRARY})
-    target_include_directories(${ARROW_LIBRARY} SYSTEM BEFORE PRIVATE ${ZLIB_INCLUDE_DIR})
-endif ()
-
-target_include_directories(${ARROW_LIBRARY} SYSTEM PRIVATE ${ORC_INCLUDE_DIR})
-target_include_directories(${ARROW_LIBRARY} SYSTEM PRIVATE ${ORC_SOURCE_SRC_DIR})
-target_include_directories(${ARROW_LIBRARY} SYSTEM PRIVATE ${ORC_SOURCE_WRAP_DIR})
-target_include_directories(${ARROW_LIBRARY} SYSTEM PRIVATE ${GOOGLE_PROTOBUF_DIR})
-target_include_directories(${ARROW_LIBRARY} SYSTEM PRIVATE ${ORC_BUILD_SRC_DIR})
-target_include_directories(${ARROW_LIBRARY} SYSTEM PRIVATE ${ORC_BUILD_INCLUDE_DIR})
-target_include_directories(${ARROW_LIBRARY} SYSTEM PRIVATE ${ORC_ADDITION_SOURCE_DIR})
-target_include_directories(${ARROW_LIBRARY} SYSTEM PRIVATE ${FLATBUFFERS_INCLUDE_DIR})
-target_include_directories(${ARROW_LIBRARY} SYSTEM PRIVATE ${HDFS_INCLUDE_DIR})
+add_dependencies(_arrow ${FLATBUFFERS_LIBRARY})
+
+target_link_libraries(_arrow PRIVATE ${FLATBUFFERS_LIBRARY} boost::filesystem)
+
+add_dependencies(_arrow protoc)
+
+target_include_directories(_arrow SYSTEM BEFORE PUBLIC ${ARROW_SRC_DIR})
+target_include_directories(_arrow SYSTEM BEFORE PUBLIC "${CMAKE_CURRENT_BINARY_DIR}/cpp/src")
+target_link_libraries(_arrow PRIVATE ch_contrib::double_conversion)
+target_link_libraries(_arrow PRIVATE ch_contrib::protobuf)
+target_link_libraries(_arrow PRIVATE ch_contrib::lz4)
+target_link_libraries(_arrow PRIVATE ch_contrib::snappy)
+target_link_libraries(_arrow PRIVATE ch_contrib::zlib)
+target_link_libraries(_arrow PRIVATE ch_contrib::zstd)
+
+target_include_directories(_arrow SYSTEM BEFORE PUBLIC  ${ORC_INCLUDE_DIR})
+target_include_directories(_arrow SYSTEM BEFORE PUBLIC  ${ORC_BUILD_INCLUDE_DIR})
+target_include_directories(_arrow SYSTEM PRIVATE ${ORC_SOURCE_SRC_DIR})
+target_include_directories(_arrow SYSTEM PRIVATE ${ORC_SOURCE_WRAP_DIR})
+target_include_directories(_arrow SYSTEM PRIVATE ${ORC_BUILD_SRC_DIR})
+target_include_directories(_arrow SYSTEM PRIVATE ${ORC_ADDITION_SOURCE_DIR})
+target_include_directories(_arrow SYSTEM PRIVATE ${ARROW_SRC_DIR})
+target_include_directories(_arrow SYSTEM PRIVATE ${FLATBUFFERS_INCLUDE_DIR})
```

Comment:
Why do these two libs not populate their only INTERFACE_INCLUDE_DIRECTORIES so that we can use modern cmake usage requirement directly? (e.g. `target_link_libraries( _arrow PRIVATE ch_contrib::flatbuffers ch_contrib::hdfs)` )

Reply:
>Why do these two libs not populate their only INTERFACE_INCLUDE_DIRECTORIES

They populate PUBLIC part (which is PRIVATE + INTERFACE), and only those, which is required for public interface.

> so that we can use modern cmake usage requirement directly? (e.g. target_link_libraries( _arrow PRIVATE ch_contrib::flatbuffers ch_contrib::hdfs) )

This is indeed what is used later.

Reply:
Oh, you are talking about doing the same thing for flatbuffers, for now it is important not to use `${LIB_INCLUDE_DIR}` outside of contrib, everything other was leaved more or less as-is.